synapseclient/models/file.py

```diff
@@ -516,40 +625,82 @@ async def store_async(
                 "The file must have an (ID with a (path or `data_file_handle_id`)), or a "
                 "(path with a (`parent_id` or parent with an id)) to store."
             )
+        self.parent_id = parent.id if parent else self.parent_id
```

Comment:
The changes in this `store_async` function serve a few purposes:

1. Retrieve the current entity data from Synapse and determine if we need to PUT new data onto Synapse
2. Determine based on the file Synapse current has if an upload needs to occur
3. Swapping over to an async/HTTPX interface for storing all of this data